assets/src/edit-story/components/richText/karma/singleSelection.karma.js

```diff
@@ -330,4 +330,42 @@ describe('Styling single text field', () => {
       expect(actual).toBe(expected);
     });
   });
+
+  describe('CUJ: Creator Can Style Text: When a font without weight support replaces font, default font weight to closest match', () => {
+    it('should replace Bold Roboto with normal weight font for Abel since Bold is not an option', async () => {
+      const {
+        fontWeight,
+      } = data.fixture.editor.inspector.designPanel.textStyle;
+      expect(fontWeight.value).toBe('Regular');
+
+      await data.fixture.events.click(fontWeight.select);
+      await data.fixture.events.click(fontWeight.option('Bold'));
+      expect(fontWeight.value).toBe('Bold');
+
+      const input = await data.fixture.screen.getByLabelText(
+        'Edit: Font family'
+      );
+      expect(input).toBeTruthy();
+      await data.fixture.events.click(input);
+
+      await data.fixture.events.keyboard.type('Ab');
+      // Ensure the debounced callback has taken effect.
+      await wait(300);
+
+      let options = document
+        .getElementById('editor-font-picker-list')
+        .querySelectorAll('li[role="option"]');
+      expect(options.length).toBe(2);
+
+      await data.fixture.events.click(options[0]);
+
+      expect(fontWeight.value).toBe('Regular');
+    });
+  });
 });
+
+function wait(ms) {
```

Comment:
Use `data.fixture.events.sleep` instead

Reply:
thanks for catching this. stole this from textStyle.cuj.karma - i'll update both places to ensure better practices moving forward.